### PR TITLE
Gradual Turn While Gliding

### PR DESCRIPTION
**Pull request by randomPoison**
- Added a gradual turn while gliding, making it feel more light and airy.
- Fixed some bugs relating to gliding.
- Removed `GroundedCheck()` from the `ActorPhysics` `FixedUpdate()` loop.

### CODE REVIEW

File: Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs [hunk 1]
```diff
@@ -179,6 +178,51 @@ public void RegisterStateMethod( ActorStates state, ActorStateMethod method )
 		}
 	}
 
+	public bool GroundedCheck()
+	{
+		_isOnGround = false;
```
Comment (brickman1444) [inline Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs:183]: maybe add a comment here explaining what the point of the spherecast is

File: Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs [hunk 2]
```diff
@@ -179,6 +178,51 @@ public void RegisterStateMethod( ActorStates state, ActorStateMethod method )
...
+		{
+			if ( !IsInState( ActorStates.Grounded ) && !IsInState( ActorStates.Rolling ) )
+			{
+				ChangeState( ActorStates.Grounded );
+				_stopMoveTimer = 0f;
+			}
+
+			actor.animator.SetBool( "isJumping", false );
+			CancelLateJumpTimer();
+			//actor.GetAnimator().SetBool("isSliding", false);
```
Comment (brickman1444) [inline Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs:208]: is this still needed?

File: Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs [hunk 3]
```diff
@@ -179,6 +178,51 @@ public void RegisterStateMethod( ActorStates state, ActorStateMethod method )
...
+		}
+		else if ( IsInState( ActorStates.Grounded ) )
+		{
+			if ( !actor.animator )
+			{
+				actor.animator.SetBool( "isJumping", true );
+				//actor.GetAnimator().SetBool("isSliding", false);
```
Comment (brickman1444) [inline Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs:215]: here too

**Comment by brickman1444** [comment]
Line 30 in PlayerControls.cs should be getting input from an axis defined in the input manager, not a hardcoded key.

**Comment by brickman1444** [comment]
The gliding feels pretty good. I think I'd reccomend we make most of the mechanics use the same button to do and just use a situation based reasoning to figure out what to do. i.e in the air, left click starts to glide, in front of vines is starts to climb, in front of a buddy it gives a resource and anywhere else it cuts.

**Comment by cjacobwade** [comment]
Combining mechanics to very few buttons is good. I think having feed/cut on left click and then glide/climb on right click would be good. On controllers it would be either trigger for climbing/gliding and then X for feeding/cutting.

File: Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs (hunk 1 above)
Comment (randomPoison) [inline Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs:183]: done.

File: Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs (hunk 2 above)
Comment (randomPoison) [inline Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs:208]: The sliding stuff? @cjacobwade Are you planning on adding sliding? Or is this a remnant from pop n pals?

**Comment by randomPoison** [comment]
I'm also cool with combining input, but that's a bit bigger than this pull request. If that's the only other issue I'll merge in this request and make a separate one for combing controls.

File: Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs (hunk 2 above)
Comment (cjacobwade) [inline Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs:208]: That's just leftover!

**Pull request merged**